ratelimit/rate-limits.go

```diff
@@ -9,15 +9,28 @@ import (
 	"github.com/letsencrypt/boulder/cmd"
 )
 
-// RateLimitConfig is an exported container for a rateLimitConfig and a mutex
-// This allows the inner rateLimitConfig pointer to be updated safely when the
-// overall configuration changes (e.g. due to a live reload of the policy file)
-type RateLimitConfig struct {
+// RateLimitConfig is defined as an interface to allow mock implementations be
+// provided during unit testing
+type RateLimitConfig interface {
+	TotalCertificates() RateLimitPolicy
+	CertificatesPerName() RateLimitPolicy
+	RegistrationsPerIP() RateLimitPolicy
+	PendingAuthorizationsPerAccount() RateLimitPolicy
+	CertificatesPerFQDNSet() RateLimitPolicy
+	LoadPolicies(contents []byte) error
+}
+
+// MutexRateLimitConfig is an exported implementation of the RateLimitConfig
+// interface. It acts as a container for a rateLimitConfig and a mutex. This
+// allows the inner rateLimitConfig pointer to be updated safely
+// when the overall configuration changes (e.g. due to a reload of the
+// policy file)
+type MutexRateLimitConfig struct {
```

Comment:
When we have four words in a type name I start to get a little nervous. Also I'm reluctant to put so much implementation detail (`Mutex`) in a type name.

How about this: The only thing that needs to actually be exported here is the interface and a `New()` function. We can also reduce the stutter in the interface name:

```
type Limits interface{
type impl {
func New() Limits {
  return &impl{}
}
```